### PR TITLE
Server tests for the management layer

### PR DESCRIPTION
**Pull request by hiendo**

### CODE REVIEW

File: management/pom.xml [hunk 1]
```diff
@@ -88,6 +120,19 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.1</version>
+                <configuration>
+                    <source>1.7</source>
```
Comment (bitwisekaizen) [inline management/pom.xml]: Why not 1.8?  Get with the times!

File: management/src/test/java/com/bitwisekaizen/sdss/acceptance/IscsiTargetManagementTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import static org.junit.Assert.fail;
+
+@Test(groups = UNIMPLEMENTED)
+public class IscsiTargetManagementTest extends AbstractAcceptanceTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(IscsiTargetManagementTest.class);
+    public static final String UNIMPLEMENTED = "unimplemented";
+
+    @AfterMethod
+    public void afterMethod() {
+        deleteAllIscsiTargets();
```
Comment (bitwisekaizen) [inline management/src/test/java/com/bitwisekaizen/sdss/acceptance/IscsiTargetManagementTest.java]: Instead, why not just keep the database in memory?

File: management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.bitwisekaizen.sdss.dto;
+
+import java.util.List;
+
+/**
+ * Spec that describes the ISCSI target
+ */
+public class IscsiTargetSpec {
```
Comment (bitwisekaizen) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java]: I think I might like to remain consistent with naming here.  IscsiTargetRequest would be consistent with PxeSessionRequest in pxeasy.  Spec is pretty general and can mean either input or output.

File: management/src/test/java/com/bitwisekaizen/sdss/dto/IscsiTargetBuilder.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import static com.bitwisekaizen.sdss.dto.IscsiTargetSpecBuilder.aIscsiTargetSpec;
+
+public class IscsiTargetBuilder implements Builder<IscsiTarget> {
+    private static final Random random = new Random();
+
+    private String uuid = UUID.randomUUID().toString();
+    private String storageIpAddress = InetAddresses.fromInteger(random.nextInt()).getHostAddress();
+    private IscsiTargetSpec iscsiTargetSpec = aIscsiTargetSpec().build();
+
+    public static IscsiTargetBuilder aIscsiTarget() {
```
Comment (bitwisekaizen) [inline management/src/test/java/com/bitwisekaizen/sdss/dto/IscsiTargetBuilder.java]: anIscsiTarget

File: management/src/test/java/com/bitwisekaizen/sdss/dto/IscsiTargetBuilder.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.UUID;
+
+import static com.bitwisekaizen.sdss.dto.IscsiTargetSpecBuilder.aIscsiTargetSpec;
+
+public class IscsiTargetBuilder implements Builder<IscsiTarget> {
+    private static final Random random = new Random();
+
+    private String uuid = UUID.randomUUID().toString();
+    private String storageIpAddress = InetAddresses.fromInteger(random.nextInt()).getHostAddress();
+    private IscsiTargetSpec iscsiTargetSpec = aIscsiTargetSpec().build();
```
Comment (bitwisekaizen) [inline management/src/test/java/com/bitwisekaizen/sdss/dto/IscsiTargetBuilder.java]: Change name of variable if classname is updated.

File: management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTarget.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package com.bitwisekaizen.sdss.dto;
+
+/**
+ * ISCSI target currently managed by the SDSS.
+ */
+public class IscsiTarget {
```
Comment (bitwisekaizen) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTarget.java]: This isn't really an iscsi target ... the iscsitargetspec is.  This is more of a entity that describes the storage service agent and all of the iscsi targets on it.  I'd call this LioBackedAgent or something.

File: management/pom.xml (hunk 1 above)
Comment (hiendo) [inline management/pom.xml]: sure

File: management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTarget.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    public String getStorageIpAddress() {
+        return storageIpAddress;
+    }
+
+    /**
+     * Get spec describing the ISCSI target.
+     *
+     * @return ISCSI target spec.
+     */
+    public IscsiTargetSpec getIscsiTargetSpec() {
```
Comment (bitwisekaizen) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTarget.java]: I'd make this many to one.

**Comment by bitwisekaizen** [comment]
Looks good.  I would like to see one more pass with the updated names just to make sure the abstractions are named appropriately.

File: management/src/test/java/com/bitwisekaizen/sdss/acceptance/IscsiTargetManagementTest.java (hunk 2 above)
Comment (hiendo) [inline management/src/test/java/com/bitwisekaizen/sdss/acceptance/IscsiTargetManagementTest.java]: this is more robust to new test adding without remember to always add to some list.. i'll forget for sure

File: management/pom.xml (hunk 1 above)
Comment (hiendo) [inline management/pom.xml]: Actually, I think I'm going to keep it at 7.  Only issue is having my mac switch between 7 and 8 for console work and this work here.

File: management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java (hunk 3 above)
Comment (hiendo) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java]: IscsiTargetSpec is also used in composition in IscsiTarget.  With your suggestion below, I'll rename this to IscsiTarget and the curren tIscsiTarget to something relating to LioBackedAgent.  Let's discuss offline though as there's also Nfs.  LioBackedIscsiAgent vs LioBackedNfsAgent?

Comment (hiendo) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java]: Does it also changed when it's backed by Nexenta?  I mean, there's still going to be a UUID, IP, and the information about the target right?  Something we can iterate on but let's discuss before we productionize this.

Comment (hiendo) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java]: Hm, kinda confusing as LioBackedIscsiAgent is not an agent.  It's additional information on the IscsiTarget.  In vSphere, there's a VirtualMachine and then within it, there's VirtualMachineConfigInfo that further describes the VM.  I was trying to do something similar with these names but can be improved on.

File: management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+    private List<String> hostIscsiQualifiedNames;
+    private int capacityInMb;
+    private String targetName;
+
+    // Json serialization
+    private IscsiTargetSpec(){}
+
+    public IscsiTargetSpec(List<String> hostIscsiQualifiedNames, int capacityInMb, String targetName) {
```
Comment (bitwisekaizen) [inline management/src/main/java/com/bitwisekaizen/sdss/dto/IscsiTargetSpec.java]: Might want to shoot for an abstraction with naming similar to the following:

IscsiTarget(IscsiAccessControlList acl, int cap, String deviceName)

IscsiTarget(IscsiACL acl, IscsiDevice device)

IscsiDevice(int cap, String deviceName)

IscsiXXXX(IscsiTarget target, String storageIp)
+ uuid:UUID

UniquelyIdentifiableIscsiTarget - target that's unique within the entire storage service.

**Pull request closed without merging**